src/neuroconv/datainterfaces/ecephys/cellexplorer/cellexplorerdatainterface.py

```diff
@@ -1,12 +1,328 @@
+
+    Detailed information can be found in the following link
+    https://cellexplorer.org/datastructure/data-structure-and-format/#channels
+
+    Future versions of this interface will support the extraction of this metadata from these files.
+    """
+
+    sampling_frequency_key = "sr"
+    binary_file_extension = "dat"
+
+    def __init__(self, folder_path: FolderPathType, verbose: bool = True, es_key: str = "ElectricalSeries"):
+        self.folder_path = Path(folder_path)
+
+        # No super here, we need to do everything by hand
+        self.verbose = verbose
+        self.es_key = es_key
+        self.subset_channels = None
+        self.source_data = dict(folder_path=folder_path)
+        self._number_of_segments = 1  # CellExplorer is mono segment
+
+        self.session = self.folder_path.name
+        session_data_file_path = self.folder_path / f"{self.session}.session.mat"
+        assert session_data_file_path.is_file(), f"File {session_data_file_path} does not exist"
+
+        from pymatreader import read_mat
+
+        ignore_fields = ["animal", "behavioralTracking", "timeSeries", "spikeSorting", "epochs"]
+        session_data = read_mat(filename=session_data_file_path, ignore_fields=ignore_fields)["session"]
+        extracellular_data = session_data["extracellular"]
+        num_channels = int(extracellular_data["nChannels"])
+        gain = float(extracellular_data["leastSignificantBit"])  # 0.195
+        gains_to_uv = np.ones(num_channels) * gain
+        dtype = np.dtype(extracellular_data["precision"])
+        sampling_frequency = float(extracellular_data[self.sampling_frequency_key])
+
+        # Channels in CellExplorer are 1-indexed
+        channel_ids = [str(1 + i) for i in range(num_channels)]
+        binary_file_path = self.folder_path / f"{self.session}.{self.binary_file_extension}"
+        if binary_file_path.is_file():
+            from spikeinterface.core.binaryrecordingextractor import (
+                BinaryRecordingExtractor,
+            )
+
+            self.recording_extractor = BinaryRecordingExtractor(
+                file_paths=[binary_file_path],
+                sampling_frequency=sampling_frequency,
+                num_chan=num_channels,
+                dtype=dtype,
+                t_starts=None,
+                file_offset=0,
+                gain_to_uV=gains_to_uv,
+                offset_to_uV=None,
+                channel_ids=channel_ids,
+            )
+        else:
+            from spikeinterface.core.numpyextractors import NumpyRecording
+
+            traces_list = [np.empty(shape=(1, num_channels))]
+            dummy_recording = NumpyRecording(
+                traces_list=traces_list,
+                sampling_frequency=sampling_frequency,
+                channel_ids=channel_ids,
+            )
+
+            self.recording_extractor = dummy_recording
+            self.recording_extractor.set_channel_gains(channel_ids=channel_ids, gains=np.ones(num_channels) * gain)
```

Comment:
Thinking more on this, I really don't think we should have this `else` block

As is, there's nothing preventing someone from writing the dummy data to an NWB file by mistake because they forgot to disable `write_electrical_series` in the conversion options

Given the work in #494 the correct approach would be to use the sorting interface instead

And if we ever do run into the situation where there are (a) no raw data files (b) they used a different sorting format (such as kilosort) and (c) but they still have valuably cellexplorer metadata files, we should really just make either a specific `MetadataInterface` that adds no data to the `NWBFile` but only metadata, or a `MetadataMixin` that allows any Sorting or Recording Interface to incorporate different metadata protocols that might result from format mixing

Reply:
I think this is a good point.

Reply:
I removed it.